test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp

```diff
@@ -99,65 +50,73 @@ void TestSnapshot::tearDown() {
-
-    SnapshotItem itemA("a", SyncDb::driveRootNode().nodeIdLocal().value(), Str("A"), 1640995201, 1640995201,
-                       NodeType::NodeTypeDirectory, 123);
-    _syncPal->_localSnapshot->updateItem(itemA);
-    CPPUNIT_ASSERT(_syncPal->_localSnapshot->exists("a"));
-    CPPUNIT_ASSERT(_syncPal->_localSnapshot->name("a") == Str("A"));
-    CPPUNIT_ASSERT(_syncPal->_localSnapshot->type("a") == NodeType::NodeTypeDirectory);
-    auto itItem = _syncPal->_localSnapshot->_items.find(SyncDb::driveRootNode().nodeIdLocal().value());
-    CPPUNIT_ASSERT(itItem->second.childrenIds().find("a") != itItem->second.childrenIds().end());
-
-    _syncPal->_localSnapshot->updateItem(SnapshotItem("a", SyncDb::driveRootNode().nodeIdLocal().value(), Str("A*"), 1640995202,
-                                                      1640995202, NodeType::NodeTypeDirectory, 123));
-    CPPUNIT_ASSERT(_syncPal->_localSnapshot->name("a") == Str("A*"));
-
-    SnapshotItem itemB("b", SyncDb::driveRootNode().nodeIdLocal().value(), Str("B"), 1640995203, 1640995203,
-                       NodeType::NodeTypeDirectory, 123);
-    _syncPal->_localSnapshot->updateItem(itemB);
-    CPPUNIT_ASSERT(_syncPal->_localSnapshot->exists("b"));
-    itItem = _syncPal->_localSnapshot->_items.find(SyncDb::driveRootNode().nodeIdLocal().value());
-    CPPUNIT_ASSERT(itItem->second.childrenIds().find("b") != itItem->second.childrenIds().end());
-
-    SnapshotItem itemAA("aa", "a", Str("AA"), 1640995204, 1640995204, NodeType::NodeTypeDirectory, 123);
-    _syncPal->_localSnapshot->updateItem(itemAA);
-    itItem = _syncPal->_localSnapshot->_items.find("a");
-    CPPUNIT_ASSERT(itItem->second.childrenIds().find("aa") != itItem->second.childrenIds().end());
-
-    SnapshotItem itemAAA("aaa", "aa", Str("AAA"), 1640995205, 1640995205, NodeType::NodeTypeFile, 123);
-    _syncPal->_localSnapshot->updateItem(itemAAA);
-    CPPUNIT_ASSERT(_syncPal->_localSnapshot->exists("aaa"));
-    itItem = _syncPal->_localSnapshot->_items.find("aa");
-    CPPUNIT_ASSERT(itItem->second.childrenIds().find("aaa") != itItem->second.childrenIds().end());
+    const DbNode dummyRootNode(0, std::nullopt, SyncName(), SyncName(), "1", "1", std::nullopt, std::nullopt, std::nullopt,
+                               NodeTypeDirectory, 0, std::nullopt);
+    Snapshot snapshot(ReplicaSideLocal, dummyRootNode);
+
+    // Insert node A
+    const SnapshotItem itemA("a", SyncDb::driveRootNode().nodeIdLocal().value(), Str("A"), 1640995201, 1640995201,
+                             NodeType::NodeTypeDirectory, 123);
+    snapshot.updateItem(itemA);
+    CPPUNIT_ASSERT(snapshot.exists("a"));
+    CPPUNIT_ASSERT_EQUAL(std::string("A"), SyncName2Str(snapshot.name("a")));
+    CPPUNIT_ASSERT_EQUAL(NodeType::NodeTypeDirectory, snapshot.type("a"));
+    auto itItem = snapshot._items.find(SyncDb::driveRootNode().nodeIdLocal().value());
```

Comment:
Why do we need to access private attributes of the `Snapshot` class?

Reply:
Done